polyply/tests/test_apply_links.py

```diff
@@ -181,112 +181,243 @@ def test_match_link_and_residue_atoms_fail(example_meta_molecule,
+                                                        meta={"version": 1})],
+                          [vermouth.molecule.Interaction(atoms=(0, 1),
+                                                        parameters=['1', '0.33', '500'],
+                                                        meta={"version": 1})],
+                          [vermouth.molecule.Interaction(atoms=(0, 1, 2, 3),
+                                                        parameters=['1', '0.33', '500'],
+                                                        meta={})]],
+                         [[], [], []],
+                         [[], [], [[[0, {'label': 'C', 'name': 'BB1'}], [1, {'label': 'A', 'name': 'BB'}],
+                                   [2, {'name': 'BB1'}], [3, {'label': 'B', 'name': 'BB2'}]]]],
+                         [[(1, 4, 1)], [(6, 9, 1)]],
+                         [[(0, 0)], [(1, 0)]]
                         ),
                         ))
 def test_apply_link_to_residue(example_meta_molecule,
                                link_defs,
                                link_to_resids,
-                               inter_type,
+                               inter_types,
                                link_inters,
+                               link_non_edges,
+                               link_patterns,
                                expected_nodes,
                                expected_inters):
     links = []
     processor = ApplyLinks()
-    for link_nodes, link_to_resid, interactions in zip(link_defs,
-                                                       link_to_resids,
-                                                       link_inters):
+    for link_nodes, link_to_resid, interactions, non_edges, patterns, inter_type in zip(link_defs,
+                                                                                        link_to_resids,
+                                                                                        link_inters,
+                                                                                        link_non_edges,
+                                                                                        link_patterns,
+                                                                                        inter_types):
         link = Link()
         link.add_nodes_from(link_nodes)
         link.interactions[inter_type] = interactions
-        processor.apply_link_between_residues(example_meta_molecule,
-                                              link,
-                                              link_to_resid)
+        link.non_edges = non_edges
+        link.patterns = patterns
+        try:
+            processor.apply_link_between_residues(example_meta_molecule,
+                                                  link,
+                                                  link_to_resid)
+        except MatchError:
+            continue
```

Comment:
Shouldn't this cause a failing test? I.e. shouldn't non-matching links get their own test to make sure they don't match?

Reply:
Yeah I guess we could do that as well